docs/src/man/reshaping_new.md

```diff
@@ -0,0 +1,498 @@
+   6 │  2020  Soap    month            6
+   7 │  2020  Fruits  month            7
+   8 │  2020  Oil     month            8
+  ⋮  │   ⋮      ⋮       ⋮         ⋮
+  18 │  2020  Soap    Date            20
+  19 │  2020  Fruits  Date            21
+  20 │  2020  Oil     Date            22
+  21 │  2021  Sugar   Date            23
+  22 │  2021  Soap    Date            24
+  23 │  2021  Fruits  Date            25
+  24 │  2021  Oil     Date            26
+                           9 rows omitted
+
+julia> long_costs = stack(costs, [:month, :Date], [:year, :list],
+                                 variable_name=:days, value_name=:costs)
+24×4 DataFrame
+ Row │ year   list    days    costs
+     │ Int64  String  String  Float64
+─────┼────────────────────────────────
+   1 │  2019  Sugar   month       0.5
+   2 │  2019  Soap    month       1.0
+   3 │  2019  Fruits  month       1.5
+   4 │  2019  Oil     month       2.0
+   5 │  2020  Sugar   month       2.5
+   6 │  2020  Soap    month       3.0
+   7 │  2020  Fruits  month       3.5
+   8 │  2020  Oil     month       4.0
+  ⋮  │   ⋮      ⋮       ⋮        ⋮
+  18 │  2020  Soap    Date       10.0
+  19 │  2020  Fruits  Date       10.5
+  20 │  2020  Oil     Date       11.0
+  21 │  2021  Sugar   Date       11.5
+  22 │  2021  Soap    Date       12.0
+  23 │  2021  Fruits  Date       12.5
+  24 │  2021  Oil     Date       13.0
+                        9 rows omitted
+
+julia> long = innerjoin(long_grocessary, long_costs, on=[:year, :list, :days])
+24×5 DataFrame
+ Row │ year   list    days    grocessary  costs
+     │ Int64  String  String  Int64       Float64
+─────┼────────────────────────────────────────────
+   1 │  2019  Sugar   month            1      0.5
+   2 │  2019  Soap    month            2      1.0
+   3 │  2019  Fruits  month            3      1.5
+   4 │  2019  Oil     month            4      2.0
+   5 │  2020  Sugar   month            5      2.5
+   6 │  2020  Soap    month            6      3.0
+   7 │  2020  Fruits  month            7      3.5
+   8 │  2020  Oil     month            8      4.0
+  ⋮  │   ⋮      ⋮       ⋮         ⋮          ⋮
+  18 │  2020  Soap    Date            20     10.0
+  19 │  2020  Fruits  Date            21     10.5
+  20 │  2020  Oil     Date            22     11.0
+  21 │  2021  Sugar   Date            23     11.5
+  22 │  2021  Soap    Date            24     12.0
+  23 │  2021  Fruits  Date            25     12.5
+  24 │  2021  Oil     Date            26     13.0
+                                    9 rows omitted
+```                                           
+
+## The basics of `stack`
+
+The concept of stacking comes in handy when we have data with multi indices. We use the `stack()` function 
+to reshape the data frame by converting the data into the stacked form. Reshaping the data from wide to 
+long format using the `stack` function means moving (also rotating or pivoting) the innermost column index 
+to become the innermost row index.
```

Comment:
It seems that this is copied from pandas. We do not have such concepts in DataFrames.jl